ansible_collections/arista/avd/roles/eos_designs/python_modules/connected_endpoints/utils.py

```
@@ -0,0 +1,287 @@
+        return get(self._hostvars, "switch.enable_trunk_groups") is True
+
+    @cached_property
+    def _platform_settings_feature_support_interface_storm_control(self) -> bool:
+        return get(self._hostvars, "switch.platform_settings.feature_support.interface_storm_control", default=True) is True
+
+    @cached_property
+    def _evpn_short_esi_prefix(self) -> str:
+        return get(self._hostvars, "evpn_short_esi_prefix", required=True)
+
+    def _get_short_esi(self, adapter: dict, channel_group_id: int, short_esi: str = None, hash_extra_value: str = "") -> str | None:
+        """
+        Return short_esi for one adapter
+        """
+        if len(set(adapter["switches"])) < 2:
+            # Only configure ESI for multi-homing.
+            return None
+
+        # short_esi is only set when called from sub-interface port-channels.
+        if short_esi is None:
+            # Setting short_esi under port_channel will be deprecated in AVD4.0
+            port_channel_short_esi = get(adapter, "port_channel.short_esi")
+            if (short_esi := get(adapter, "ethernet_segment.short_esi", default=port_channel_short_esi)) is None:
+                return None
+
+        endpoint_ports: list = default(
+            adapter.get("endpoint_ports"),
+            adapter.get("server_ports"),
+        )
+        short_esi = str(short_esi)
+        if short_esi.lower() == "auto":
+            esi_hash_switches = "".join(adapter["switches"][:2])
+            esi_hash_switch_ports = "".join(adapter["switch_ports"][:2])
+            esi_hash_endpoint_ports = "".join(endpoint_ports[:2])
+            esi_hash = sha256(
+                f"{hash_extra_value}{esi_hash_switches}{esi_hash_switch_ports}{esi_hash_endpoint_ports}{channel_group_id}".encode("UTF-8")
+            ).hexdigest()
+            short_esi = re.sub(r"([0-9a-f]{4})", "\\1:", esi_hash)[:14]
+
+        if len(short_esi.split(":")) != 3:
+            return None
```

Are we sure of this? if I give a `short_esi` to this method with a wrong format I guess I would prefer to be told? I know this is the current behavior to silently ignore but ..

---

The format feature of schemas should catch this - but not there yet.

---

added raise